.github/workflows/CI.yml

```diff
@@ -63,38 +66,36 @@ jobs:
           pipenv install --deploy --dev
           pipenv run test
           pipenv run coverage xml
-      - name: Upload Coverage
-        uses: codecov/codecov-action@v4
+      - name: Get Sanitized Name
+        env:
+          PY_PROJECT: ${{ matrix.py-project }}
+        run: echo "SANITIZED_PY_PROJECT=${PY_PROJECT/\//-}" >> $GITHUB_ENV
+      - name: Upload Coverage Reports
+        uses: actions/upload-artifact@v4
         with:
-          fail_ci_if_error: true
-          files: ./coverage.xml
-          flags: python-test
-          token: ${{ secrets.CODECOV_TOKEN }}
-          verbose: true
+          if-no-files-found: error
+          name: coverage-reports-python-tests-${{ env.SANITIZED_PY_PROJECT }}
+          path: ${{ env.WORKING_DIR }}/coverage.xml
```

Comment:
Seems strange to reference the working directory - isn't `coverage.xml` enough?

Reply:
This task is run from the root and ignores the setting
```
defaults:
      run:
        working-directory
```

Reply:
I could instead use `./**/coverage.xml`, but it seemed more accurate to just point to the file we're expecting